packages/sit-onyx/src/components/OnyxDataGrid/features/index.ts

```diff
@@ -49,15 +64,17 @@ export type TypeRenderer<TEntry extends DataGridEntry> = {
 export type TypeRenderMap<
   TEntry extends DataGridEntry,
   TKey extends PropertyKey = PropertyKey,
-> = Partial<Record<TKey, TypeRenderer<TEntry>>>;
+> = Partial<Record<TKey, Readonly<TypeRenderer<TEntry, object>>>>;
 
 /**
  * ColumnConfig as it can be defined by the user.
  */
 export type ColumnConfig<
   TEntry extends DataGridEntry,
-  TColumnGroup extends ColumnGroupConfig,
-  TTypes,
+  TColumnGroup extends ColumnGroupConfig = Record<string, never>,
+  TTypes extends ColumnConfigTypeOption<PropertyKey, unknown> = RenderTypesFromFeature<
```

Comment:
Can we merge this type with the default types? Otherwise, when defining custom types like in our "CustomColumnTypes" example, all the default types will be removed
![image](https://github.com/user-attachments/assets/cff6b3f6-79bf-4d96-a057-bc2f6f0a5ce3)